src/Resolver/AbstractInjection.php

```diff
@@ -7,6 +7,19 @@
 
 namespace Zend\Di\Resolver;
 
+use const E_USER_DEPRECATED;
+use function trigger_error;
+
+trigger_error(
+    __CLASS__ . ' is deprecated, please migrate to ' . InjectionInterface::class,
```

Comment:
I don't think `__CLASS__` will be defined at this point. Use `AbstractInjection::class` here instead.

Also, since you're using a couple different variable/constant strings, I'd use `sprintf()` here.

Reply:
> I don't think `__CLASS__` will be defined at this point.

Indeed, I'll update the test as well...